src/Interpreters/Context.h

```diff
@@ -1456,6 +1468,24 @@ class Context: public ContextData, public std::enable_shared_from_this<Context>
     std::pair<Context::StorageMetadataCache *, std::unique_lock<std::mutex>> getStorageMetadataCache() const;
     std::pair<Context::StorageSnapshotCache *, std::unique_lock<std::mutex>> getStorageSnapshotCache() const;
 
+     /// Information needed in pure index functions (without real column access).
+    std::pair<IndexContextInfoPtr, std::shared_lock<std::shared_mutex>> getIndexInfo() const
```

Comment:
Maybe add a simple comment that functions can access this info in parallel but only for reading purpose. Also, maybe return `std::pair<const IndexContextInfoPtr, ...` as another hint that it's not modifiable